content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,834 @@
+1. Open `VegetationBending_Common.azsli`.
+
+1. At the bottom, add a function to return the world position of the vertex after applying bending. The parameters given to this function are all helpful for determining bending.
+
+   ```hlsl
+   float4 ProcessBending(float currentTime, float3 objectSpacePosition, float3 normal, float4 detailBendingParams, float4 worldPosition, float4x4 objectToWorld) 
+   {
+      float4 adjustedWorldPosition = float4(worldPosition);
+      if (o_color_isBound) 
+      {
+         // We will add function calls here.
+      }
+      return adjustedWorldPosition;
+   }
+   ```
+
+   {{< note >}}
+   Notice how, like before, we are using a conditional with `o_color_isBound` to ensure that we only perform bending on proper meshes.
+   {{< /note >}}
+
+1. Call the `ProcessBending` function in our vertex shaders. Open `VegetationBending_ForwardPass.azsli` and find the vertex shader (`VegetationBending_ForwardPassVS`). Add a call to our function above `OUT.m_worldPosition = worldPosition.xyz`. Most of the parameters to pass in are inputs to our vertex shader or values we have calculated already, and we can get the time from the property `m_time` the *scene shader resource group* (`SceneSrg`), which gives us the number of seconds since the start of the application.
+   
+   ```hlsl
+   float currentTime = SceneSrg::m_time;
+   worldPosition = ProcessBending(currentTime, IN.m_position, IN.m_normal, IN.m_optional_color, worldPosition, objectToWorld);
+
+   OUT.m_worldPosition = worldPosition.xyz;
+   OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, worldPosition);
+
+   return OUT;
+   ``` 
+   {{< note >}}
+   We need the code to be above the two `OUT` lines because we will edit `worldPosition` and then will need to adjust the `OUT` variables accordingly.
+   {{< /note >}}
+  
+1. Repeat steps 2-3 with the depth pass in `VegetationBending_DepthPass.azsli` and the depth pass with PS in `VegetationBending_DepthPass_WithPS.azsli`.
+
+   {{< tip >}}
+   Note that the depth pass does not output a world position. Just place the code above the `OUT.m_position` line. The depth pass with PS has some extra code in the vertex shader, but you can still place this code about the two `OUT` statements.
+   {{< /tip >}}
+
+### Set up wind bending
+Let's begin editing the code to add wind. 
+
+1. Open `VegetationBending_Common.azsli`.
+
+1. At the bottom, add a function to calculate the amplitude, frequency, and phase of the wind according to the time and world position of the vertex. The phase should take into account the `worldPosition` because we want to mimic how wind should affect nearby objects similarly but faraway objects differently. Faraway objects may not be affected by the same "breeze". This function will then later be used to calculate the appropriate movement of the vertex.
```

Comment:
Add new lines to break apart this step

```suggestion
1. At the bottom, add a function to calculate the amplitude, frequency, and phase of the wind according to the time and world position of the vertex. The wind's phase uses the `worldPosition` to mimic how wind affects nearby objects similarly, but faraway objects differently. This is because in real life, faraway objects may not be affected by the same breeze. 

  Later, we'll use this function to calculate the appropriate movement of the vertex.
```